atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/knowledge/ConcludedRangeState.java

```diff
@@ -23,17 +23,24 @@
 import com.google.common.collect.Range;
 import com.google.common.collect.RangeSet;
 import com.google.common.collect.Sets;
+import com.palantir.atlasdb.transaction.impl.TransactionConstants;
 import java.util.Set;
 import org.immutables.value.Value;
 
 @Value.Immutable
-@JsonSerialize(as = ImmutableTimestampRangeSet.class)
-@JsonDeserialize(as = ImmutableTimestampRangeSet.class)
+@JsonSerialize(as = ImmutableConcludedRangeState.class)
+@JsonDeserialize(as = ImmutableConcludedRangeState.class)
 @SuppressWarnings("UnstableApiUsage") // RangeSet usage
-public interface TimestampRangeSet {
+public interface ConcludedRangeState {
 
+    /**
+     * The minimum timestamp is used to ensure we do not conclude ranges that are below this timestamp [-∞, ts).
+     * This is necessary for backup/restore, as we must not conclude any ranges post-restore that are before
+     * the fast-forward timestamp. This is to prevent concluding transactions which have written to the KVS,
+     * but had not committed.
+     */
     @Value.Parameter
-    Long minimumTimestamp();
+    Long minimumConcludeableTimestamp();
 
```

Comment:
still would like it better if we can move the filtering logic down here because then no one outside needs to even access it.